### PR TITLE
bws-95: * added backend functionality for update group

### PR DESCRIPTION
**Pull request by bohdanshcherba**

### CODE REVIEW

File: backend/src/api/eam/eam.api.ts [hunk 1]
```diff
@@ -91,6 +92,35 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
     },
   });
 
+  fastify.route({
+    method: HttpMethod.POST,
```
Comment (what1s1ove) [inline backend/src/api/eam/eam.api.ts]:
```suggestion
    method: HttpMethod.PUT,
```

File: backend/src/data/repositories/group/group.repository.ts [hunk 2]
```diff
@@ -60,42 +60,103 @@ class Group {
...
-      permissionsIds.map((permissionId) => ({
+      permissionIds.map((permissionId) => ({
         id: getRandomId(),
         groupId: id,
         permissionId: permissionId,
-        createdAt: createdAt,
+        createdAt: new Date().toISOString(),
       })),
     );
+  }
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/group/group.repository.ts]: Move these methods to GroupEntity:
```
in service:

const group = await repo.getGroupById()
group.addWorkersids();
await repo.save(group)
```

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/api/eam/eam.api.ts [hunk 3]
```diff
@@ -91,6 +92,35 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
+    },
+
+    async handler(
+      req: FastifyRequest<{
+        Querystring: { id: string };
+        Body: EAMGroupCreateRequestDto;
+      }>,
+      rep,
+    ) {
+      const group = await groupService.update(req.query.id, req.body);
```
Comment (what1s1ove) [inline backend/src/api/eam/eam.api.ts]:
```suggestion
      const group = await groupService.update(req.params.id, req.body);
```

File: backend/src/api/eam/eam.api.ts [hunk 4]
```diff
@@ -91,6 +92,35 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
+      return (
+        data: EAMGroupCreateRequestDto,
+      ): ReturnType<typeof groupConfigurateValidationSchema['validate']> => {
+        return schema.validate(data);
+      };
+    },
+
+    async handler(
+      req: FastifyRequest<{
+        Querystring: { id: string };
```
Comment (what1s1ove) [inline backend/src/api/eam/eam.api.ts]:
```suggestion
```
take it from params

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/eam/eam.api.ts [hunk 5]
```diff
@@ -92,6 +93,36 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
+      return (
+        data: EAMGroupCreateRequestDto,
+      ): ReturnType<typeof groupConfigurateValidationSchema['validate']> => {
+        return schema.validate(data);
+      };
+    },
+
+    async handler(
+      req: FastifyRequest<{
+        Params: { id: string };
```
Comment (what1s1ove) [inline backend/src/api/eam/eam.api.ts:114]: create a separate type for it
check in other similar routes

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/group/group.entity.ts [hunk 6]
```diff
@@ -76,6 +76,18 @@ class Group {
...
+    this.name = name ? name : this.name;
+  }
+
+  public setWorkersIds(workersIds: string[]): void {
+    this.workersIds = workersIds ? workersIds : this.workersIds;
+  }
+
+  public setPermissionsIds(permissionsIds: string[]): void {
+    this.permissionsIds = permissionsIds ? permissionsIds : this.permissionsIds;
+  }
```
Comment (Yaroslaww-1) [inline backend/src/services/group/group.entity.ts:90]: Why do we need this checks? I think it's better to remove them.
If you want to validate input data it's always better to explicitly throw an exception

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Comment by bohdanshcherba** [comment]
@Yaroslaww-1 can you close this branch without merge?

**Pull request closed without merging**